### ISSUE TITLE
"ghc-pkg: command not found" when starting flok-repl -t tidal

### ISSUE THREAD

**Issue by santoluigi** [munshkr/flok#83]
hi

when i try to use flock (web browser) with tidal, it doesn't work. 
when i tried to connect REPLs to Flok i get this error:
 
Error: Command failed: ghc-pkg field tidal data-dir
/bin/sh: ghc-pkg: command not found

how i can fix this issue?
thanks

**Comment by jwaldmann** [comment]
how/where did you install ghc? what is the output of
```
which ghc
```
normally, `ghc-pkg` is in the very same directory.

See also #85 (the shell finds ghc-pkg, but ghc-pkg does not find tidal)

**Comment by santoluigi** [comment]
ok the `which ghc` output is:

`/Users/username/.ghcup/bin/ghc`

but if I do `cabal list tidal` the output is:

`* maquinitas-tidal
    Synopsis: library for MIDI control of hardware
    Default available version: 0.2.4
    Installed versions: [ Not installed ]
    Homepage: https://github.com/maquinitas/maquinitas-tidal
    License:  MIT

* tidal
    Synopsis: Pattern language for improvised music
    Default available version: 1.6.1
    Installed versions: [ Not installed ]
    Homepage: http://tidalcycles.org/
    License:  GPL-3

* tidal-midi
    Synopsis: MIDI support for tidal
    Default available version: 0.9.10
    Installed versions: [ Not installed ]
    Homepage: http://tidal.lurk.org/
    License:  GPL-3

* tidal-serial
    Synopsis: Serial support for tidal
    Default available version: 0.8
    Installed versions: [ Not installed ]
    Homepage: http://tidalcycles.org/
    License:  GPL-3

* tidal-vis
    Synopsis: Visual rendering for Tidal patterns and osc messages
    Default available version: 1.0.15
    Installed versions: [ Not installed ]
    Homepage: http://yaxu.org/tidal/
    License:  GPL-3
`
but tidal is installed because I use it with atom.

how I can fix the issue? 
i saw the post  but #85 but didn't understand how it was fixed and if it is ok for this problem

**Comment by jwaldmann** [comment]
You reported `ghc-pkg: command not found`. That's strange. When I install ghc and tools via `ghcup`, it also installs `$HOME/.ghcup/bin/ghc-pkg`. Do you have this? If yes, then we need to debug why the `flok` repl does not find it. Perhaps `$PATH` not set correctly?

**Comment by santoluigi** [comment]
ok this is the complete output when I try to do a flak session:
now the problem seems the path of tidal package?

`Hub address: wss://flok.clic.cf
Session name: MTM2YTQ5NjQtMWQxYS00ODUwLWIwZGMtMDcwNjdlZjljNDMy
Target name: tidal
Type: tidal
Extra options: {}
ghc-pkg: cannot find package tidal
Failed to get tidal data-dir
You will need to specify location of TidalCycles bootloading script.
Read more: https://github.com/munshkr/flok/wiki/Failed-to-get-tidal-data-dir
/usr/local/lib/node_modules/flok-repl/lib/repl/tidal.js:37
            throw err;
            ^

Error: Command failed: ghc-pkg field tidal data-dir
ghc-pkg: cannot find package tidal

    at checkExecSyncError (child_process.js:630:11)
    at Object.execSync (child_process.js:666:15)
    at TidalREPL.dataDir (/usr/local/lib/node_modules/flok-repl/lib/repl/tidal.js:27:45)
    at TidalREPL.defaultBootScript (/usr/local/lib/node_modules/flok-repl/lib/repl/tidal.js:22:31)
    at TidalREPL.bootScript (/usr/local/lib/node_modules/flok-repl/lib/repl/tidal.js:19:35)
    at new TidalREPL (/usr/local/lib/node_modules/flok-repl/lib/repl/tidal.js:10:43)
    at Object.<anonymous> (/usr/local/lib/node_modules/flok-repl/bin/flok-repl.js:75:16)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32) {
  status: 1,
  signal: null,
  output: [
    null,
    Buffer(0) [Uint8Array] [],
    Buffer(35) [Uint8Array] [
      103, 104,  99,  45, 112, 107, 103,  58,
       32,  99,  97, 110, 110, 111, 116,  32,
      102, 105, 110, 100,  32, 112,  97,  99,
      107,  97, 103, 101,  32, 116, 105, 100,
       97, 108,  10
    ]
  ],
  pid: 7759,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: Buffer(35) [Uint8Array] [
    103, 104,  99,  45, 112, 107, 103,  58,
     32,  99,  97, 110, 110, 111, 116,  32,
    102, 105, 110, 100,  32, 112,  97,  99,
    107,  97, 103, 101,  32, 116, 105, 100,
     97, 108,  10
  ]
}
`

**Comment by munshkr** [comment]
Hi @santoluigi 

The problem is that flok-repl tries to use `ghc-pkg` to find the location of Tidal package in your system, because it needs to find the Tidal bootstrap script (`BootTidal.hs`). Unfortunately, there was a change in the way ghc or cabal handles this and this doesn't work anymore on recent GHC versions.

In the meantime, you can force `flok-repl` to use a specific `BootTidal.hs`, and that will circumvent the ghc-pkg issue.
First you need to find yourself the path to your BootTidal.hs. An easy way would be to search everywhere in your home directory maybe:

```
find ~ -name BootTidal.hs 
```

This will list every appearance of a file named `BootTidal.hs`. You should look for one that refers to the Tidal version you installed. If you are in doubt, please print the output and we can help you.

Then, just add this at the end of your flok-repl command:
```
--extra '{ "bootScript": "/home/munshkr/Documents/Tidal/BootTidal.hs" }'
```
You need to change the path there with the one you selected from the `find` output.

**Comment by munshkr** [comment]
Alternatively, you can download BootTidal.hs from [here](https://raw.githubusercontent.com/tidalcycles/Tidal/main/BootTidal.hs). This is the latest version, so use this if you installed the latest version of Tidal.

**Comment by munshkr** [comment]
In the end I decided to include the latest BootTidal.sh as a fallback, in case ghc-pkg fails to get tidal data package, so this should be fixed in the next flok-repl v0.4.1
